airflow/providers/cncf/kubernetes/decorators/kubernetes.py

```diff
@@ -82,15 +84,35 @@ def _get_python_source(self):
         res = remove_task_decorator(res, "@task.kubernetes")
         return res
 
+    def _generate_cmds(self):
+        return [
+            "bash",
+            "-cx",
+            (
+                f"{_generate_decoded_command('%s', '%s')} && " % (_PYTHON_SCRIPT_ENV, _FILENAME_IN_CONTAINER)
```

Comment:
Why do you need to pass in placeholders and immediately resolve them, instead of passing in the values directly…?

Reply:
Just wondering if you know of any character limits to `cmds` in K8s? That's the only concern I have here because long functions can have very long b64 contents.

Reply:
Nothing I can find [here](https://kubernetes.io/docs/tasks/inject-data-application/define-command-argument-container/). Seems to work with a lot of gibberish...

<img width="817" alt="image" src="https://user-images.githubusercontent.com/9200263/212697701-e5cf6592-968e-4ed0-97fc-f4d0183eac6a.png">

<img width="1121" alt="image" src="https://user-images.githubusercontent.com/9200263/212697743-348aea0a-effe-4908-85f9-2445c1c08b92.png">

Reply:
What I am asking is, instead of

```python
f"{_generate_decoded_command('%s', '%s')} && " % (_PYTHON_SCRIPT_ENV, _FILENAME_IN_CONTAINER)
```

Why can’t you write

```python
f"{_generate_decoded_command(_PYTHON_SCRIPT_ENV, _FILENAME_IN_CONTAINER)} &&"
```

?

Reply:
Also I realised as I was writing the above comment, naively wrapping the value in single quotes is not a good idea. You likely want to use `shlex.quote`.

Reply:
Ah I see. Sorry, misunderstood. Reverting previous commit and will try it this way.